src/java_tools/buildjar/java/com/google/devtools/build/buildjar/BazelJavaBuilder.java

```diff
@@ -43,10 +43,15 @@ public static void main(String[] args) {
       WorkRequestHandler workerHandler = new WorkRequestHandler(builder::parseAndBuild);
       System.exit(workerHandler.processRequests(System.in, System.out, System.err));
     } else {
-      System.exit(
-          builder.parseAndBuild(
-              Arrays.asList(args),
-              new PrintWriter(new OutputStreamWriter(System.err, Charset.defaultCharset()))));
+      PrintWriter pw =
```

Comment:
Can you use try-with-resources?

Reply:
The goal is to call `flush()` in the finally not `close()` (which would impolitely close the underlying stderr).

Reply:
Doesn't `close` also call `flush`? And don't you just call `System.exit()` right after? So it might actually be polite to close stderr before :)

Reply:
Sorry, you're right, System.exit is too abrupt to be handled by try-with-resources.

Reply:
Yes, currently it doesn't matter much because the program promptly exits, but it is a refactoring hazard.

Anyway, I don't feel strongly about this, so I'll just autoclose.

Reply:
I agree we don't want to close `System.err`, otherwise if something goes wrong between closing it and exiting we'll get a context-free failure. Calling `flush` is the right thing to do.

Reply:
Sorry @benjaminp again. I stand corrected.